<h1> Clubhouse Dataset </h1>
Clubhouse is one of the social media platform where social interaction is done through live, audio-only conversation. Clubhouse allows users to talk and listen in real time. users can join a conversation in a room full of people or start a virtual room to talk to each other. No recordings options was provided so everything talked about was on real-time.

 A user *[johntukey](https://www.kaggle.com/datasets/johntukey/clubhouse-dataset)* has made this dataset publicly available through kaggle.<br>
 Firstly,let's import the dataset and necessary libraries.

In [1]:
import sqlite3
import pandas as pd

Since the data in the dataset are in .db format which is database binary format, we need a sql reader to read those data. Here we have imported sqlite3 and pandas to read those data.

In [2]:
db = sqlite3.connect('/kaggle/input/clubhouse-dataset/Clubhouse_Dataset_v1.db')
df = pd.read_sql_query('SELECT * FROM user', db)

Here I have used sqlite3.connect() function to establish a connection to the given database. Next, I have used pandas read_sql_query() function to read the data in the database. we need to pass two arguments to use this function, so I passed a SQL query to retrieve all user's info and the database connection that I created above. Now lets look at some data.

In [4]:
print(df.info(5))

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1300515 entries, 0 to 1300514
Data columns (total 10 columns):
 #   Column                   Non-Null Count    Dtype 
---  ------                   --------------    ----- 
 0   user_id                  1300515 non-null  int64 
 1   name                     1300515 non-null  object
 2   photo_url                1236637 non-null  object
 3   username                 1300515 non-null  object
 4   twitter                  1300515 non-null  object
 5   instagram                1300515 non-null  object
 6   num_followers            1300515 non-null  int64 
 7   num_following            1300515 non-null  int64 
 8   time_created             1300515 non-null  object
 9   invited_by_user_profile  1300515 non-null  object
dtypes: int64(3), object(7)
memory usage: 99.2+ MB
None


info() function in pandas will give us an overview of the data contained in this file. Like a total number of users, the datatype of each column, memory usage, etc.

In [8]:
print(df.head())

   user_id              name  \
0        4        Rohan Seth   
1        5      Paul Davison   
2        8  Johnny Appleseed   
3       10              DK 🖍   
4       12  Jonathan Gheller   

                                           photo_url  username    twitter  \
0  https://clubhouseprod.s3.amazonaws.com:443/4_b...     rohan  rohanseth   
1  https://clubhouseprod.s3.amazonaws.com:443/5_e...      paul   pdavison   
2                                               None    apple1       null   
3  https://clubhouseprod.s3.amazonaws.com:443/10_...        dk       dksf   
4  https://clubhouseprod.s3.amazonaws.com:443/12_...  jonathan   jgheller   

      instagram  num_followers  num_following  \
0          null        4187268            599   
1          null        3718334           1861   
2  srt_tester_9             20             81   
3          null          49538            173   
4          null          21250             81   

                       time_created invited_by_us

Here, I have used the head() function from pandas, which gives us information about the first 5 data in the dataset. As we can see, these data are scattered, so we will try to display them correctly in the next step.

In [9]:
from IPython.core.display import display, HTML

display(
    HTML(
        "<div style='height: fit-content; overflow: auto; width: fit-content'>" +
        df.head(10).to_html() +
        "</div>"
    )
)

Here, I have used the IPython library to display data in a single line for each user using iPython library. Next, we will try to display these data in chunks so that we can display much more data without wasting memory. The dataset is large with **1300515** entries; we cannot display all values. 

In [7]:
chunk_size = 10

for chunk in pd.read_sql_query('SELECT * FROM user', db, chunksize =  chunk_size):
    display(chunk)
    input("press enter to continue ...")

db.close()

user_id              name  \
0        4        Rohan Seth   
1        5      Paul Davison   
2        8  Johnny Appleseed   
3       10              DK 🖍   
4       12  Jonathan Gheller   
5       13             bubba   
6       14   Todd Goldberg 🍜   
7       15  Yan-David Erlich   
8       16       Merci Grace   
9       17     Jordan Cooper   

                                           photo_url      username  \
0  https://clubhouseprod.s3.amazonaws.com:443/4_b...         rohan   
1  https://clubhouseprod.s3.amazonaws.com:443/5_e...          paul   
2                                               None        apple1   
3  https://clubhouseprod.s3.amazonaws.com:443/10_...            dk   
4  https://clubhouseprod.s3.amazonaws.com:443/12_...      jonathan   
5  https://clubhouseprod.s3.amazonaws.com:443/13_...         bubba   
6  https://clubhouseprod.s3.amazonaws.com:443/14_...          todd   
7  https://clubhouseprod.s3.amazonaws.com:443/15_...         yanda   
8  https://clubhouseprod.s3.amazonaws.com:443/16_...         merci   
9  https://clubhouseprod.s3.amazonaws.com:443/17_...  jordancooper   

        twitter     instagram  num_followers  num_following  \
0     rohanseth          null        4187268            599   
1      pdavison          null        3718334           1861   
2          null  srt_tester_9             20             81   
3          dksf          null          49538            173   
4      jgheller          null          21250             81   
5         bubba          null          47428            266   
6      toddg777          null          15725            389   
7         yanda          null           5054            313   
8         merci          null          56916            158   
9  jordancooper          null           4741            190   

                       time_created invited_by_user_profile  
0  2020-03-17T07:51:28.085566+00:00                    null  
1  2020-03-17T14:36:19.468976+00:00                    null  
2  2020-03-19T19:47:00.323603+00:00                    null  
3  2020-03-19T23:38:52.574777+00:00                    null  
4  2020-03-20T02:30:22.188084+00:00                    null  
5  2020-03-20T02:32:24.778265+00:00                    null  
6  2020-03-21T17:22:08.522772+00:00                    null  
7  2020-03-21T19:08:46.826227+00:00                    null  
8  2020-03-21T19:25:07.681258+00:00                    null  
9  2020-03-22T01:26:02.100146+00:00                    null

press enter to continue ... 


user_id                       name  \
0       18  Christian Sutherland-Wong   
1       19          Christine Raschke   
2       20                  Steve Lee   
3       22                 Rohin Dhar   
4       23                 Chris Paik   
5       24                 hunterwalk   
6       25                Matt Mazzeo   
7       26               M.G. Siegler   
8       27                      Naval   
9       28             Talia Goldberg   

                                           photo_url    username  \
0  https://clubhouseprod.s3.amazonaws.com:443/18_...   christian   
1  https://clubhouseprod.s3.amazonaws.com:443/19_...   christine   
2  https://clubhouseprod.s3.amazonaws.com:443/20_...   moneyball   
3  https://clubhouseprod.s3.amazonaws.com:443/22_...       rohin   
4  https://clubhouseprod.s3.amazonaws.com:443/23_...       cpaik   
5  https://clubhouseprod.s3.amazonaws.com:443/24_...  hunterwalk   
6  https://clubhouseprod.s3.amazonaws.com:443/25_...      mazzeo   
7  https://clubhouseprod.s3.amazonaws.com:443/26_...         mgs   
8  https://clubhouseprod.s3.amazonaws.com:443/27_...       naval   
9  https://clubhouseprod.s3.amazonaws.com:443/28_...       talia   

        twitter     instagram  num_followers  num_following  \
0  christian_sw          null           1006            156   
1      craschke  sissiraschke          21188           1488   
2     moneyball          null          10637            295   
3          null          null            255            128   
4         cpaik          null           3905            247   
5    hunterwalk          null          35834            133   
6        Mazzeo       mmazzeo          23276            338   
7     mgsiegler     mgsiegler          19274            373   
8          null          null         617280            332   
9     TaliaGold          null           2532            149   

                       time_created invited_by_user_profile  
0  2020-03-22T02:23:55.946564+00:00                    null  
1  2020-03-22T02:44:39.571262+00:00                    null  
2  2020-03-22T05:10:59.167610+00:00                    null  
3  2020-03-23T01:16:59.509688+00:00                    null  
4  2020-03-23T01:58:23.559922+00:00                    null  
5  2020-03-23T03:48:35.563926+00:00                    null  
6  2020-03-23T04:53:54.891287+00:00                    null  
7  2020-03-23T15:22:11.396276+00:00                    null  
8  2020-03-23T15:54:32.224712+00:00                    null  
9  2020-03-23T21:54:41.445340+00:00                    null

press enter to continue ... 


user_id              name  \
0       30    Charles Hudson   
1       31     David Hegarty   
2       32        James Cham   
3       33                Om   
4       35  Jonathan Ehrlich   
5       37       Harj Taggar   
6       39     Catherine Lee   
7       40    Jason Crawford   
8       41     James Beshara   
9       42        Mike Duboe   

                                           photo_url       username  \
0  https://clubhouseprod.s3.amazonaws.com:443/30_...        chudson   
1  https://clubhouseprod.s3.amazonaws.com:443/31_...           hegs   
2  https://clubhouseprod.s3.amazonaws.com:443/32_...      jamescham   
3  https://clubhouseprod.s3.amazonaws.com:443/33_...             om   
4  https://clubhouseprod.s3.amazonaws.com:443/35_...            jon   
5  https://clubhouseprod.s3.amazonaws.com:443/37_...           harj   
6  https://clubhouseprod.s3.amazonaws.com:443/39_...        cathlee   
7  https://clubhouseprod.s3.amazonaws.com:443/40_...  jasoncrawford   
8  https://clubhouseprod.s3.amazonaws.com:443/41_...          james   
9  https://clubhouseprod.s3.amazonaws.com:443/42_...           mike   

           twitter instagram  num_followers  num_following  \
0          chudson      null          47312            242   
1             null      null            236            121   
2        jamescham      null           8154            269   
3             null      null           8832            197   
4  jonathanehrlich      null           3902            130   
5       harjtaggar      null           6376             63   
6             null      null            173             81   
7    jasoncrawford      null          30672            411   
8     jamesbeshara      null           8730             67   
9           mduboe      null           9441            231   

                       time_created invited_by_user_profile  
0  2020-03-23T23:08:43.940833+00:00                    null  
1  2020-03-24T00:59:50.254757+00:00                    null  
2  2020-03-24T06:11:58.290680+00:00                    null  
3  2020-03-24T15:00:46.595975+00:00                    null  
4  2020-03-25T00:28:37.028289+00:00                    null  
5  2020-03-25T01:31:24.686366+00:00                    null  
6  2020-03-25T14:51:49.472018+00:00                    null  
7  2020-03-25T20:56:12.607126+00:00                    null  
8  2020-03-25T20:59:15.908383+00:00                    null  
9  2020-03-25T21:37:01.168305+00:00                    null

press enter to continue ... 


user_id              name  \
0       43     Siobhan Quinn   
1       44    Sahil Lavingia   
2       45  Jonathan Wegener   
3       46    Aaron Batalion   
4       48            Yin Wu   
5       49     Brian Donohue   
6       50     Gregory Rosen   
7       51   D’Arcy Coolican   
8       52    Amanda Davison   
9       54          Elad Gil   

                                           photo_url       username  \
0  https://clubhouseprod.s3.amazonaws.com:443/43_...        siobhan   
1  https://clubhouseprod.s3.amazonaws.com:443/44_...          sahil   
2  https://clubhouseprod.s3.amazonaws.com:443/45_...       jwegener   
3  https://clubhouseprod.s3.amazonaws.com:443/46_...  aaronbatalion   
4  https://clubhouseprod.s3.amazonaws.com:443/48_...            yin   
5  https://clubhouseprod.s3.amazonaws.com:443/49_...     bthdonohue   
6  https://clubhouseprod.s3.amazonaws.com:443/50_...             gr   
7  https://clubhouseprod.s3.amazonaws.com:443/51_...      dcoolican   
8  https://clubhouseprod.s3.amazonaws.com:443/52_...         amanda   
9  https://clubhouseprod.s3.amazonaws.com:443/54_...           elad   

        twitter     instagram  num_followers  num_following  \
0  siobhanquinn  siobhanquinn           1495            231   
1           shl     slavingia         219472            405   
2      jwegener      jwegener           3937            116   
3     abatalion          null          18358            392   
4          null          null            686             80   
5    bthdonohue          null           5773            388   
6        grosen          null           4448            195   
7     DCoolican          null          11853            151   
8   adavison001          null           3071           2425   
9          null          null          39866            304   

                       time_created invited_by_user_profile  
0  2020-03-25T22:18:04.158983+00:00                    null  
1  2020-03-25T23:09:49.775884+00:00                    null  
2  2020-03-25T23:51:02.968150+00:00                    null  
3  2020-03-26T04:45:38.932419+00:00                    null  
4  2020-03-26T05:50:47.378439+00:00                    null  
5  2020-03-26T18:37:03.846471+00:00                    null  
6  2020-03-26T18:37:04.047263+00:00                    null  
7  2020-03-26T18:59:06.584801+00:00                    null  
8  2020-03-26T19:39:26.181869+00:00                    null  
9  2020-03-26T23:34:38.662208+00:00                    null

press enter to continue ... 


user_id                 name  \
0       55      Lenny Rachitsky   
1       56    nikita singareddy   
2       57  Nikhil Basu Trivedi   
3       58           Steve Jang   
4       59         Avichal Garg   
5       60            Ben Rubin   
6       61         Nabeel Hyatt   
7       62   Jana Messerschmidt   
8       65           Josh Elman   
9       66          David Tisch   

                                           photo_url username       twitter  \
0  https://clubhouseprod.s3.amazonaws.com:443/55_...    lenny      lennysan   
1  https://clubhouseprod.s3.amazonaws.com:443/56_...   nikita  singareddynm   
2  https://clubhouseprod.s3.amazonaws.com:443/57_...      nbt           nbt   
3  https://clubhouseprod.s3.amazonaws.com:443/58_...       sj     stevejang   
4  https://clubhouseprod.s3.amazonaws.com:443/59_...  avichal       avichal   
5  https://clubhouseprod.s3.amazonaws.com:443/60_...   benrbn        benrbn   
6  https://clubhouseprod.s3.amazonaws.com:443/61_...   nabeel        nabeel   
7  https://clubhouseprod.s3.amazonaws.com:443/62_...  janamal       janamal   
8  https://clubhouseprod.s3.amazonaws.com:443/65_...     josh     joshelman   
9  https://clubhouseprod.s3.amazonaws.com:443/66_...    david    davidtisch   

   instagram  num_followers  num_following                      time_created  \
0   lennysan          13555            239  2020-03-26T23:45:04.657638+00:00   
1       null           8951             94  2020-03-27T00:07:54.922044+00:00   
2   nikhilbt          16223            362  2020-03-27T03:06:41.864180+00:00   
3  stevejang          19165            769  2020-03-27T04:16:30.750648+00:00   
4       null          35208            311  2020-03-27T20:21:07.416358+00:00   
5     benrbn           3335            294  2020-03-27T20:29:51.236436+00:00   
6    nabeelo          17715            246  2020-03-28T21:27:49.040953+00:00   
7    janamal          82153            111  2020-03-28T22:09:28.564824+00:00   
8       null          71676            904  2020-03-29T23:22:41.089702+00:00   
9       null          17927            545  2020-03-29T23:25:19.316043+00:00   

  invited_by_user_profile  
0                    null  
1                    null  
2                    null  
3                    null  
4                    null  
5                    null  
6                    null  
7                    null  
8                    null  
9                    null

KeyboardInterrupt: Interrupted by user